### PR TITLE
Remove unnecessary memcpy() calls in src/utils.c

### PR DESCRIPTION
**Pull request by y-guyon**
See https://en.cppreference.com/w/c/language/object#Strict_aliasing

Following https://github.com/AOMediaCodec/libavif/pull/882#discussion_r827426932

### CODE REVIEW

File: src/utils.c [hunk 1]
```diff
@@ -16,45 +16,40 @@ float avifRoundf(float v)
...
-    memcpy(&result, data, sizeof(uint16_t));
     return result;
 }
 
 uint16_t avifNTOHS(uint16_t s)
 {
-    uint8_t data[2];
-    memcpy(data, &s, sizeof(data));
-
+    uint8_t* data = (uint8_t*)&s;
```
Comment (wantehchang) [inline src/utils.c:28]: In the NTOH functions, `data` should be declared as a const pointer.

**Review by wantehchang — CHANGES_REQUESTED** [review]
Thank you for the pull request. Please run clang-format (or an equivalent method) on this file.

**Comment by y-guyon** [comment]
> Thank you for the pull request. Please run clang-format (or an equivalent method) on this file.

Sorry, I forgot again. Switching between code bases and code styles...

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**